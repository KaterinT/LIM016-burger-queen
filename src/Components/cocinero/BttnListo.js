import { updateDoc, doc } from "firebase/firestore";
import { db } from "../../firebase.config";

export const TemplatePedidos = ({ objetoPedido }) => {
    return <>
            <p><b>Hora:</b> {objetoPedido.hora}</p>
            <p><b>Cliente: </b> {objetoPedido.cliente}</p>
            <ul>
              <b>Pedidos: </b>
              {objetoPedido.pedidosArray.map((e) => (<li key={e}>{e}</li>))}
            </ul>
          </>
  };
  
  export const PedidoEstadoFalse = ({objeto}) => {
    const  reconocimiento = (e) => {
        updateDoc(doc(db,'ordenes',e.target.name),{estado:true});
    }
    return <>
            <div className="infoMesa"> <b>Mesa: </b>{objeto.mesa}</div>
            <div className="infoPedido">
                <TemplatePedidos objetoPedido={objeto} />
                <button name={objeto.id} onClick={reconocimiento}>Listo?</button>
            </div>
           </>
  }