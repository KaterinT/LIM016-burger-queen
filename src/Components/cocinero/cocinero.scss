@import url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@1,300&display=swap');

.pedidos-al-chef{
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    button{
        font-family: Roboto;
        font-style: italic;
        font-size: 22px;
        padding: 0.8vh 1.5%;
        cursor: pointer;
    }
    #toDo{
        border-radius: 20px 0px 0px 0px;
    }
    #done{
        border-radius: 0px 20px 0px 0px;
    }
}
#eventosChef{
    overflow: auto;
    margin: 0 4%;
    padding: 1%;
    height: 58vh;
    background: rgba(0, 0, 0, 0.55);
    border: 1px solid #FFD809;
    box-sizing: border-box;
    display: flex;
    flex-wrap: wrap;
}
#eventosChef::-webkit-scrollbar {
    width: 10px;
  }
  
#eventosChef::-webkit-scrollbar-thumb {
    background: #FFD809;
    border-radius: 5px;
  }
  
#eventosChef::-webkit-scrollbar-thumb:active {
    background: #c5a60d;
  }
.pedidos{
    margin: 1%;
    height: 75%;
    padding: 1%;
    width: 21%;
    background: rgba(245, 196, 123, 0.36);
    border-radius: 7px;
    color: blanchedalmond;
    font-family: Roboto;
    font-size: 18px;
    position: relative;
    .infoPedido p{
        margin: 3px;
    }
    .infoPedido button{
        background: #FDD70B;
        border-radius: 10px;
        width: 26%;
        margin: 0 30%;
        position: absolute;
        bottom: 4%;
        color: rgb(0, 0, 0);
    }
    .infoPedido ul {
        margin-top: 0;
        margin-left: 8%;
        padding: 1%;
    }
}
.infoMesa{
    color: black;
    font-size: 17px;
    background: #F1C375;
    border-radius: 0px 7px;
    top: 0;
    position: absolute;
    right: 0;
    padding: 2px;
}
.clicked{
    border: 2px solid #854F25;
    background-color:#FFD809;
    color: black;
    height: 7vh;
    border-color: #FFD809;
}
.no-clicked{
    border: 2px solid #854F25;
    background-color:rgba(0, 0, 0, 0.55);
    color: #F1C375;
}