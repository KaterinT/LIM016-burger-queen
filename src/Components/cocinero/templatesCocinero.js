

export const TemplatePedidos = ({ objetoPedido }) => {
    return <>
            <p><b>Hora:</b> {objetoPedido.hora}</p>
            <p><b>Cliente: </b> {objetoPedido.cliente}</p>
            <ul>
              <b>Pedidos: </b>
              {objetoPedido.pedidosArray.map((e) => (<li key={e}>{e}</li>))}
            </ul>
          </>
  };
  
  export const PedidoEstadoFalse = ({objeto, cambioEstado}) => {
    
    return <>
            <div className="infoMesa"> <b>Mesa: </b>{objeto.mesa}</div>
            <div className="infoPedido">
                <TemplatePedidos objetoPedido={objeto} />
                <button name={objeto.id} onClick={cambioEstado}>Listo?</button>
            </div>
           </>
  }