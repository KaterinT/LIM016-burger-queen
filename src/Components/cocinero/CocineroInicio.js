import "./cocinero.scss";
import React, { useEffect, useState } from "react";
import { TemplatePedidos } from "./templatesCocinero";
import { updateDoc, doc} from "firebase/firestore";
import { db } from "../../firebase.config";
import { actualizarDoc } from "../../data/listaProductos";

/* <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
 Aqui empieza la vista COCINERO
>>>>>>>>>>>>>>>>>>>>>>>> >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>< */

export const Cocinero = ({orders,horaAc}) => {
  
  const [ordenesFiltradas, setOrdenesFilter] = useState();
  const [[bttnToDo, bttnDone], setNameClass] = useState(["clicked","no-clicked"]);
  const [estadoOrdenes, setEstado] = useState(false);
  

  useEffect(() => {
    return () => {
      setOrdenesFilter([]);
    };
  }, []);

  useEffect(() => {
    setOrdenesFilter(orders.filter((obj) => obj.estado === estadoOrdenes));
  }, [estadoOrdenes, orders]);

  const handleToDo = () => {
    setNameClass(["clicked", "no-clicked"]);
    setEstado(false);
  };

  const handleDone = () => {
    // Agregando estilos a los botones seleccionados
    setNameClass(["no-clicked", "clicked"]);
    setEstado(true);
  };
  const cambioEstadoOrden = (id) => {
    actualizarDoc('ordenes',id,{ estado: true, horaEntrega:horaAc });
    //updateDoc(doc(db, "ordenes",id), { estado: true, horaEntrega:horaAc });
    setEstado(false);
  };

  return (
    <div id="cocinero">
      <p className="horaAc">{horaAc}</p>
      <h4>Cocinero</h4>
      <section className="pedidos-al-chef">
        <button onClick={handleToDo} id="toDo" className={bttnToDo}>
          <b>Por Preparar</b>
        </button>
        <button onClick={handleDone} id="done" className={bttnDone}>
          <b>Preparados</b>
        </button>
      </section>
      <section id="eventosChef">
        {ordenesFiltradas !== undefined &&
          ordenesFiltradas.map((objeto) => (
            <div key={objeto.id} id={objeto.id} className="pedidos">
              <TemplatePedidos
                objeto={objeto}
                cambioEstado={cambioEstadoOrden}
              />
            </div>
          ))}
      </section>
    </div>
  );
};
