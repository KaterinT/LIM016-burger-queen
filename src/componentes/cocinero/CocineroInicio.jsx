/*eslint-disable */
import './cocinero.scss';
import React, { useEffect, useState } from 'react';
import { TemplatePedidos } from './templatesCocinero';
import { actualizarDoc } from '../../data/funcionesFirestore';

/* <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
 Aqui empieza la vista COCINERO
>>>>>>>>>>>>>>>>>>>>>>>> >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>< */

export function Cocinero({ orders, horaAc }) {
  const [ordenesFiltradas, setOrdenesFilter] = useState();
  const [estadoOrdenes, setEstado] = useState(false);

  useEffect(() => () => {
    setOrdenesFilter([]);
  }, []);

  useEffect(() => {
    setOrdenesFilter(orders.filter((obj) => obj.estado === estadoOrdenes));
  }, [estadoOrdenes, orders]);

  const handleToDo = () => {
    setEstado(false);
  };

  const handleDone = () => {
    setEstado(true);
  };
  const cambioEstadoOrden = (id) => {
    actualizarDoc('ordenes', id, { estado: true, horaEntrega: horaAc });
  };

  return (
    <div id="cocinero">
      <p className="horaAc">{horaAc}</p>
      <h4>Cocinero</h4>
      <section className="pedidos-al-chef">
        <button onClick={handleToDo} id="toDo" className={(estadoOrdenes === false) ? 'clicked' : 'no-clicked'}>
          <b>Por Preparar</b>
        </button>
        <button onClick={handleDone} id="done" className={(estadoOrdenes === true) ? 'clicked' : 'no-clicked'}>
          <b>Preparados</b>
        </button>
      </section>
      <section id="eventosChef">
        {ordenesFiltradas !== undefined
          && ordenesFiltradas.map((objeto) => (
            <div key={objeto.id} id={objeto.id} className="pedidos">
              <TemplatePedidos
                objeto={objeto}
                cambioEstado={cambioEstadoOrden}
              />
            </div>
          ))}
      </section>
    </div>
  );
}
