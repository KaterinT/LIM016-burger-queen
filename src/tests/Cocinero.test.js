/*eslint-disable */
import { screen, cleanup, render, fireEvent } from "@testing-library/react";
import { Cocinero } from "../componentes/cocinero/CocineroInicio";


jest.mock('../componentes/cocinero/templatesCocinero',() => ({ TemplatePedidos: ({objeto, cambioEstado}) => {
    return (<div key ={objeto.id}>
        <p>{objeto.cliente}</p>
        <button onClick={() => cambioEstado(objeto.id)}>{(objeto.estado===false)?'falso':'true'}</button>
    </div>)
} }) );
jest.mock('../data/funcionesFirestore')


describe('Empecemos a testear vista Cocinero',() => {
    const ordenes = [{
        id:1,
        data:() => {
            const objeto = {
                cliente:'usuario11',
                estado:true,
                hora:'6:14:35PM',
                mesa:1,
                pedidosArray:[
                    {descripcion:'Cafe Americano', count:2, precio:10, id:1100}
                ]
            }
            return objeto
        }
    },{
        id:2,
        data:() => {
            const objeto = {
                cliente:'usuario21',
                estado:false,
                hora:'3:10:02PM',
                mesa:1,
                pedidosArray:[
                    {descripcion:'Cafe Americano', count:4, precio:10, id:1100}
                ]
            }
            return objeto
        }
    },{
        id:3,
        data:() => {
            const objeto = {
                cliente:'usuario32',
                estado:false,
                hora:'8:10:02AM',
                mesa:2,
                pedidosArray:[
                    {descripcion:'Hamburguesa doble res', count:1, precio:15, id:2200+'res'}
                ]
            }
            return objeto
        }
    },{
        id:4,
        data:() => {
            const objeto = {
                cliente:'usuario43',
                estado:true,
                hora:'3:59:02PM',
                mesa:3,
                pedidosArray:[
                    {descripcion:'Hamurguesa simple pollo', count:2, precio:11, id:3300+'pollo'}
                ]
            }
            return objeto
        }
    }]
    const orders =ordenes.map((objeto)=> ({...objeto.data(),id:objeto.id}))
    beforeAll((done) => {
        document.body.innerHTML = '';
        done();
      });
    afterEach(cleanup);

    it('Dberiamos renderizar Cocinero' , () => {
        const {container}=render(<Cocinero orders={orders} horaAc={'2:35:00pm'}/>)
        expect(container).toMatchSnapshot();
    })
    //--------Estado inicial :falso......
    it('Deberian aparecer usuario21 y usuario32',() => {
        render(<Cocinero orders={orders} horaAc={'2:35:00pm'}/>)
        expect(screen.getByText('usuario21')).toBeInTheDocument();
        expect(screen.getByText('usuario32')).toBeInTheDocument();
    });
    it('Deberia aparecen la clase del boton Por Hacer como clicked y de Preparados como no-clicked',() => {
        render(<Cocinero orders={orders} horaAc={'2:35:00pm'}/>)
        const arrayClassname=[];
        arrayClassname.push(screen.getAllByRole('button')[0].className);
        arrayClassname.push(screen.getAllByRole('button')[1].className);

        expect(arrayClassname).toEqual(['clicked','no-clicked']);
    });
    // -------Estado true
    it('Deberia aparecen la clase del boton Por Hacer como no-clicked y de Preparados como clicked', () => {
        render(<Cocinero orders={orders} horaAc={'2:35:00pm'}/>)
        fireEvent.click(screen.getAllByRole('button')[1]);
        const arrayClassname=[];
        arrayClassname.push(screen.getAllByRole('button')[0].className);
        arrayClassname.push(screen.getAllByRole('button')[1].className);

        expect(arrayClassname).toEqual(['no-clicked','clicked']);
        expect(screen.getByText('usuario11')).toBeInTheDocument();
        expect(screen.getByText('usuario43')).toBeInTheDocument();
    })


})