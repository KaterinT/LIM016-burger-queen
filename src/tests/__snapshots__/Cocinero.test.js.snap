// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empecemos a testear vista Cocinero Dberiamos renderizar Cocinero 1`] = `
<div>
  <div
    id="cocinero"
  >
    <p
      class="horaAc"
    >
      2:35:00pm
    </p>
    <h4>
      Cocinero
    </h4>
    <section
      class="pedidos-al-chef"
    >
      <button
        class="clicked"
        id="toDo"
      >
        <b>
          Por Preparar
        </b>
      </button>
      <button
        class="no-clicked"
        id="done"
      >
        <b>
          Preparados
        </b>
      </button>
    </section>
    <section
      id="eventosChef"
    >
      <div
        class="pedidos"
        id="2"
      >
        <div>
          <p>
            usuario21
          </p>
          <button>
            falso
          </button>
        </div>
      </div>
      <div
        class="pedidos"
        id="3"
      >
        <div>
          <p>
            usuario32
          </p>
          <button>
            falso
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;
